wordpress-mobile/WordPress-Android
# RI: Add new field tags to tbl_posts

Parent Issue: #12028

We need to display tags on the discover post card in an expandable tags view as part of **Reader Improvements - Phase 2** changes. [Figma design](https://www.figma.com/file/BL53gDs6URD0yRUmGRCbhr/%F0%9F%93%90-Discover-Postcard-UI-(Specs)?node-id=492%3A659).

This PR adds new field `tags` to `tbl_posts`. The field will be used to store comma separated slugs for tags included in a post. 

- `tags` for post are parsed [here](https://github.com/wordpress-mobile/WordPress-Android/blob/4ae02c362b791d4a3c72131c9960c4822b2b402f/WordPress/src/main/java/org/wordpress/android/models/ReaderPost.java#L306-L312).
- `tags` are saved into database [here](https://github.com/wordpress-mobile/WordPress-Android/commit/2d341dea7f744887218030f6058c78234ade8964#diff-e46f5035ae37fb0d85f4f2729f5afe50R889) and retrieved from database + set to post [here](https://github.com/wordpress-mobile/WordPress-Android/commit/2d341dea7f744887218030f6058c78234ade8964#diff-e46f5035ae37fb0d85f4f2729f5afe50R1118-R1122).

**To Test**
1. Install build from `develop` on an emulator
2. Launch app and go to Reader -> Discover tab
3. Access `wpreader.db` using Device File Explorer
4. Verify that `tbl_posts` does not contain `tags` field
5. Install build from this branch on the same emulator (do not delete existing app data)
6. Enable RI Phase 2 FF
7. Go to Reader -> Discover tab
(Steps 6-7 will ensure that [these lines from `ReaderDiscoverLogic`](https://github.com/wordpress-mobile/WordPress-Android/blob/5ae5f1fcd61ddc7a49980cb0ecda7751ec86f16d/WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt#L124-L125) are executed which will parse and save tags for discover posts)
```
    val cards = parseCards(fullCardsJson)
    insertPostsIntoDb(cards.filterIsInstance<ReaderPostCard>().map { it.post })
```
8. Access `wpreader.db` using Device File Explorer
9. Verify that `tbl_posts` contains `tags` field with comma separated tags for posts.

**Open Questions**
1. Should we parse only discover posts tags? Currently haven't put any check for it. 
2. ~~Should we save entire tag to `tbl_tags`? There's an[ existing logic ](https://github.com/wordpress-mobile/WordPress-Android/blob/a3988b1b990862b2e7ee16df6737c449a43349f1/WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtils.java#L216-L240)to get a tag using its slug. Unless we want to save tag names in specific locale, we might not want to save each tag to `tbl_tags`.~~ 
~~`links` in a post tag do not seem to be of the format used for `tbl_tags.endpoint`~~
  UPDATE:  Discussed how it is being done in iOS yesterday and decided not to save entire tag to `tbl_tags`.

**Merge Instructions**
I'm targeting 15.6 milestone for this PR so that we get enough time to test. 

- Wait for 15.5 code-freeze
- Remove Not Ready for Merge label
- Merge this PR


PR submission checklist:
